src/core/thread/mle_router.cpp

```diff
@@ -707,7 +707,7 @@ void MleRouter::HandleLinkRequest(RxInfo &aRxInfo)
         {
             neighbor = mRouterTable.FindRouterByRloc16(sourceAddress);
             VerifyOrExit(neighbor != nullptr, error = kErrorParse);
-            VerifyOrExit(!neighbor->IsStateLinkRequest(), error = kErrorAlready);
+            VerifyOrExit(!neighbor->IsStateLinkRequest() || neighbor->GetRouterId() < mRouterId, error = kErrorAlready);
```

Comment:
Nice find and observation. This is even more important since we fixed and enhanced the code to add a random delay before sending a "Link Request" triggered by receiving a multicast advertisement (which many neighbors may receive and act on).

Breaking it with `neighbor->GetRouterId() < mRouterId` seems reasonable.

One small suggestion is to change the check to the following, which is semantically the same but may read better:

```c++
if (neighbor->IsStateLinkRequest())
{
    VerifyOrExit(neighbor->GetRouterId() < mRouterId, error = kErrorAlready);
}
```

I like this guideline (and try to apply it) about positive boolean check: https://testing.googleblog.com/2023/10/improve-readability-with-positive.html

Reply:
I'm fine with the proposed style.

I was wondering whether we should just remove the check, so that it will have better backward compatibility. see the comment: https://github.com/openthread/openthread/pull/11030#issuecomment-2533531472

Reply:
I slightly prefer the approach in the PR (using router ID) due to fewer msg exchanges.

Reply:
Submitted https://github.com/openthread/openthread/pull/11045 related to this.

With the above, I would say it may be better to fully remove the `VerifyOrExit(!neighbor->IsStateLinkRequest(), error = kErrorAlready)` check (less likely to have multiple exchanges) Thoughts?

Reply:
SGTM